tests/terraform/runner/test_runner.py

```diff
@@ -1015,6 +1015,43 @@ def test_record_relative_path_with_abs_file(self):
             # no need to join with a '/' because the TF runner adds it to the start of the file path
             self.assertEqual(record.repo_file_path, f'/{file_rel_path}')
 
+    @mock.patch.dict(os.environ, {"CHECKOV_ENABLE_NESTED_MODULES": "False"})
+    def test_record_definition_context_path(self):
+        resources_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "resources", "definition_context_path_nested_modules")
+        checks_allow_list = ['CKV_AWS_20']
+        main_path = os.path.join(resources_path, 'main.tf')
+        expected_definition_context_paths = [main_path,
+                                             f'{os.path.join(resources_path, "module/main.tf")}[{main_path}#0]',
+                                             f'{os.path.join(resources_path, "module/module2/main.tf")}[{main_path}->{os.path.join(resources_path, "module/main.tf")}#0]']
+        expected_definition_context_paths.sort()
+
+        runner = Runner()
+        report = runner.run(root_folder=resources_path, external_checks_dir=None,
+                            runner_filter=RunnerFilter(framework=["terraform"], checks=checks_allow_list))
+        definition_context_paths = [f.definition_context_file_path for f in report.failed_checks]
+        definition_context_paths.sort()
+        self.assertEqual(len(expected_definition_context_paths), 3)
```

Comment:
did you mean to check `definitio_context_paths`?  I don't see you editing `expected_defitiion_context_paths` anywhere...